packages/web-ui-registration/src/ResetPassword/ResetPasswordPage.tsx

```diff
@@ -0,0 +1,107 @@
+import { Button, Field, Modal, Box, Throbber, PasswordInput, InputBoxSkeleton } from '@rocket.chat/fuselage';
+import type { TranslationKey } from '@rocket.chat/ui-contexts';
+import { useRouteParameter, useRoute, useUser, useMethod, useTranslation, useLoginWithToken } from '@rocket.chat/ui-contexts';
+import type { ReactElement } from 'react';
+import { Form } from '@rocket.chat/layout';
+import { useForm } from 'react-hook-form';
+
+import HorizontalTemplate from '../template/HorizontalTemplate';
+import { usePasswordPolicy } from '../hooks/usePasswordPolicy';
+
+const getChangePasswordReason = ({
+	requirePasswordChange,
+	requirePasswordChangeReason = requirePasswordChange ? 'You_need_to_change_your_password' : 'Please_enter_your_new_password_below',
+}: { requirePasswordChange?: boolean; requirePasswordChangeReason?: TranslationKey } = {}): TranslationKey => requirePasswordChangeReason;
+
+const ResetPasswordPage = (): ReactElement => {
+	const user = useUser();
+	const t = useTranslation();
+	const setUserPassword = useMethod('setUserPassword');
+	const resetPassword = useMethod('resetPassword');
+	const token = useRouteParameter('token');
+
+	const policies = usePasswordPolicy({
+		token: user ? undefined : token,
+	});
+
+	const router = useRoute('home');
```

Comment:
I don't get why people use `router` so often. 😢

```suggestion
	const homeRoute = useRoute('home');
```

Reply:
😽 

> I don't get why people use `router` so often. 😢